qmulticast/utils/multipartite_network.py

```diff
@@ -205,3 +139,67 @@ def create_multipartite_network(name: str, graph: DiGraph) -> Network:
             mem_position += 2
 
     return network
+
+
+def add_mulitpartite_components(node,node_name,port_size,source_delay,source_noise,state_sampler):
+    logger.debug(f"Adding QSource for node 'qsource-{node_name}'.")
+
+    qsource = QSource(
+        name=f"qsource-{node_name}",
+        state_sampler=state_sampler,
+        models={
+            "emission_delay_model": source_delay,
+            "emissions_noise_model": source_noise,
+        },
+        num_ports=port_size,# size of GHZ state? each qubit goes to a out port
+        status=SourceStatus.EXTERNAL,
+    )
+    node.add_subcomponent(qsource)
+    # add first qubit to local mem
+    qsource.ports["qout0"].connect(
+        node.subcomponents["qmemory"].ports[f"qin{0}"]
+    )
+
+
+def add_non_unique_multipartite(node, node_name, graph , losses, network):
+    # node_name = node.name
+    logger.debug(f"Node: {node_name}")
+    (fibre_delay,fibre_loss,depolar_noise) = losses
```

Comment:
You pass this in to the other function and construct it in this one, better to be consistent

Reply:
changed both to pass losses